posthog/settings/web.py

```diff
@@ -50,24 +54,35 @@
     # ok below the above middlewares however.
     "posthog.health.healthcheck_middleware",
     "posthog.middleware.ShortCircuitMiddleware",
+    "django.contrib.sessions.middleware.SessionMiddleware",
+    "django.contrib.auth.middleware.AuthenticationMiddleware",
+    "django.contrib.messages.middleware.MessageMiddleware",
+]
+
+ALWAYS_POSTFIX_MIDDLEWARES = [
+    "posthog.middleware.CHQueries",
+    "django_prometheus.middleware.PrometheusAfterMiddleware",
+]
+
+OPTIONAL_MIDDLEWARE = [
     "posthog.middleware.AllowIPMiddleware",
     "google.cloud.sqlcommenter.django.middleware.SqlCommenter",
-    "django.contrib.sessions.middleware.SessionMiddleware",
     "corsheaders.middleware.CorsMiddleware",
```

Comment:
Do we have tests for Cors on capture/decide btw? I assume we need it.

Reply:
Good point, there are tests, I wonder why they didn't fail 🤔  - maybe because it's an implicit test, will fix if so.